src/rf/apps/core/tests.py

```diff
@@ -3,6 +3,141 @@
 from __future__ import unicode_literals
 from __future__ import division
 
+from django.contrib.auth.models import User
 from django.test import TestCase
 
-# Create your tests here.
+from rest_framework.test import APIClient
+
+class UserLayerTestCase(TestCase):
+    def setUp(self):
+        self.c = APIClient()
+
+        self.logged_in_user = 'bob'
+        self.other_user = 'steve'
+        self.invalid_user = 'mike'
+
+        # Create a public and private layer for each user.
+        for username in [self.logged_in_user, self.other_user]:
+            self.setup_user(username)
+
+        self.c.login(username=self.logged_in_user, password=self.logged_in_user)
+
+    def setup_user(self, username):
+        self.create_user(username)
+        self.c.login(username=username, password=username)
+        layer_name = username + ' Public Layer'
+        self.create_layer(layer_name, username, is_public = True)
+
+        layer_name = username + ' Private Layer'
+        self.create_layer(layer_name, username)
+        self.c.logout()
+
+    def create_user(self, username):
+        email = '%s@%s.com' % (username,username)
+        user = User.objects.create_user(username=username,
+                                        email=email,
+                                        password=username)
+
+    def create_layer(self, layer_name, username, is_public = False):
+        layer = {
+            'name': layer_name,
+            'is_public': is_public,
+            'capture_start': '2015-08-15',
+            'capture_end': '2015-08-15',
+            'layer_images': [
+                {'source_uri': 'http://www.foo.com'},
+                {'source_uri': 'http://www.bar.com'},
```

Comment:
I was able to flatten the tags as you suggested, but I don't think it's going to work for layer_images, because each layer_image has many other fields. The URL is just the only one without a default value.